client/service_create.go

```diff
@@ -3,28 +3,48 @@ package client
 import (
 	"encoding/json"
 
+	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/swarm"
 	"golang.org/x/net/context"
 )
 
 // ServiceCreate creates a new Service.
 func (cli *Client) ServiceCreate(ctx context.Context, service swarm.ServiceSpec, options types.ServiceCreateOptions) (types.ServiceCreateResponse, error) {
-	var headers map[string][]string
+	var (
+		headers     map[string][]string
+		manifestErr error
+	)
 
 	if options.EncodedRegistryAuth != "" {
 		headers = map[string][]string{
 			"X-Registry-Auth": {options.EncodedRegistryAuth},
 		}
 	}
 
+	// Call the /manifest endpoint to retrieve manifest information
+	if options.QueryRegistry {
+		manifestInspect, manifestErr := cli.ImageManifest(ctx, service.TaskTemplate.ContainerSpec.Image, options.EncodedRegistryAuth)
+		if manifestErr == nil {
+			// manifest was successfully retrieved, now pin by digest if the
+			// image doesn't already contain a digest
+			if _, err := reference.ParseNamed(service.TaskTemplate.ContainerSpec.Image); err != nil {
```

Comment:
Ah okay, let me add that. I wasn't going by the word "canonical" as much as the string check, but it's embarrassing now because that doesn't make any sense. Fixing it now.